env/composite_env_wrapper.h

```diff
@@ -0,0 +1,1230 @@
+// Copyright (c) 2019-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <rocksdb/env.h>
+#include <rocksdb/file_system.h>
+
+namespace rocksdb {
+
+// The CompositeEnvWrapper class provides a interface that is compatible
+// with the old monolithic Env API, and an implementation that wraps around
+// the new Env that provides threading and other OS related functionality, and
+// the new FileSystem API that provides storage functionality. By
+// providing the old Env interface, it allows the rest of RocksDB code to
+// be agnostic of whether the underlying Env implementation is a monolithic
+// Env or an Env + FileSystem. In the former case, the user will specify
+// Options::env only, whereas in the latter case, the user will specify
+// Options::env and Options::file_system.
+#define IO_STATUS_TO_STATUS(_status_, _io_status_)                 \
```

Comment:
Can we inline this and avoid using macro?

Reply:
This macro can be removed. The wrapper functions can directly return the status, since Status is now the base class of IOStatus.